modules/geoipbackend/geoipbackend.cc

```diff
@@ -85,12 +85,12 @@ GeoIPBackend::GeoIPBackend(const string& suffix)
   WriteLock writeLock(&s_state_lock);
   setArgPrefix("geoip" + suffix);
   if (!getArg("dnssec-keydir").empty()) {
-    DIR* dir = opendir(getArg("dnssec-keydir").c_str());
-    if (dir == nullptr) {
+    auto dirHandle = std::unique_ptr<DIR, decltype(&closedir)>(opendir(getArg("dnssec-keydir").c_str()), closedir);
+    if (!dirHandle) {
       throw PDNSException("dnssec-keydir " + getArg("dnssec-keydir") + " does not exist");
     }
     d_dnssec = true;
-    closedir(dir);
+    dirHandle.reset();
```

Comment:
Isn't `dirHandle` going out of scope here? Why do you need the call to `reset()`?

Reply:
It's a good question: I tried to keep my changes restricted to the smart pointer move as much as possible to avoid any side-effects, but I agree with you in this case it is probably useless. I'll take a second look, thanks!